### PR TITLE
Add span and txn id for logs derived from otel spans

### PR DESCRIPTION
**Pull request by lahsivjar**
## Motivation/summary

See https://github.com/elastic/apm-data/issues/78

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

1. Send data to an APM Server instance using [cmd/sendotlp](https://github.com/elastic/apm-server/tree/main/systemtest/cmd/sendotlp)
2. Assert that `span.id` field is populated for event with `processor.name: "log"`

## Related issues

https://github.com/elastic/apm-data/issues/78

### CODE REVIEW

File: changelogs/head.asciidoc [hunk 1]
```diff
@@ -14,6 +14,7 @@ https://github.com/elastic/apm-server/compare/8.8\...main[View commits]
 - Limit the amount of concurrent OTLP requests being processed in parallel {pull}10987[10987].
 - Added CA certificates bundle to the Docker images {pull}11015[11015]
 - Derivation of transaction and span type from OTLP spans is now deterministic {pull}11036[11036]
+- Populate span.id and transaction.id for logs dervied from OTLP spans {pull}11038[11038]
```
Comment (dmathieu) [inline changelogs/head.asciidoc:17]: How about tagging the attributes?

```suggestion
- Populate `span.id` and `transaction.id` for logs dervied from OTLP spans {pull}11038[11038]
```

Comment (dmathieu) [inline changelogs/head.asciidoc:17]:
```suggestion
- Populate span.id and transaction.id for logs derived from OTLP spans {pull}11038[11038]
```

File: changelogs/head.asciidoc [hunk 2]
```diff
@@ -14,6 +14,7 @@ https://github.com/elastic/apm-server/compare/8.8\...main[View commits]
 - Limit the amount of concurrent OTLP requests being processed in parallel {pull}10987[10987].
 - Added CA certificates bundle to the Docker images {pull}11015[11015]
 - Derivation of transaction and span type from OTLP spans is now deterministic {pull}11036[11036]
+- Populate `span.id` and `transaction.id` for logs dervied from OTLP spans {pull}11038[11038]
```
Comment (dmathieu) [inline changelogs/head.asciidoc:17]:
```suggestion
- Populate `span.id` and `transaction.id` for logs derived from OTLP spans {pull}11038[11038]
```

**Review by dmathieu — APPROVED** [review]
Sorry, I thought I had approved earlier.

**Comment by axw** [comment]
Verified with 8.9.0-BC1 using sendotlp. Navigated to the Logs tab of a trace samle, and verified the `span.id` field was present:

image

**Pull request merged**